docs/source/fx.md

```diff
@@ -0,0 +1,1442 @@
+transformation. There may be a quick answer in
+{ref}`Writing Transformations`, but, if not, there are several ways to
+examine our traced module:
+
+```
+# Sample Module
+class M(torch.nn.Module):
+    def forward(self, x, y):
+        return x + y
+
+# Create an instance of `M`
+m = M()
+
+# Symbolically trace an instance of `M` (returns a GraphModule). In
+# this example, we'll only be discussing how to inspect a
+# GraphModule, so we aren't showing any sample transforms for the
+# sake of brevity.
+traced = symbolic_trace(m)
+
+# Print the code produced by tracing the module.
+print(traced)
+# The generated `forward` function is:
+"""
+def forward(self, x, y):
+    add = x + y;  x = y = None
+    return add
+"""
+
+# Print the internal Graph.
+print(traced.graph)
+# This print-out returns:
+"""
+graph():
+    %x : [num_users=1] = placeholder[target=x]
+    %y : [num_users=1] = placeholder[target=y]
+    %add : [num_users=1] = call_function[target=operator.add](args = (%x, %y), kwargs = {})
+    return add
+"""
+
+# Print a tabular representation of the internal Graph.
+traced.graph.print_tabular()
+# This gives us:
+"""
+opcode         name    target                   args    kwargs
+-------------  ------  -----------------------  ------  --------
+placeholder    x       x                        ()      {}
+placeholder    y       y                        ()      {}
+call_function  add     <built-in function add>  (x, y)  {}
+output         output  output                   (add,)  {}
+"""
+```
+
+Using the utility functions above, we can compare our traced Module
+before and after we've applied our transformations. Sometimes, a
+simple visual comparison is enough to trace down a bug. If it's still
+not clear what's going wrong, a debugger like `pdb` can be a good
+next step.
+
+Going off of the example above, consider the following code:
+
+```
+# Sample user-defined function
+def transform_graph(module: torch.nn.Module, tracer_class : type = fx.Tracer) -> torch.nn.Module:
+    # Get the Graph from our traced Module
+    g = tracer_class().trace(module)
+
+    """
+    Transformations on `g` go here
+    """
+
+    return fx.GraphModule(module, g)
+
+# Transform the Graph
+transformed = transform_graph(traced)
+
+# Print the new code after our transforms. Check to see if it was
+# what we expected
+print(transformed)
+```
+
+Using the above example, let’s say that the call to `print(traced)`
+showed us that there was an error in our transforms. We want to find
+what goes wrong using a debugger. We start a `pdb` session. We can see
+what’s happening during the transform by breaking on
+`transform_graph(traced)`, then pressing `s` to “step into” the call
+to `transform_graph(traced)`.
+
+We may also have good luck by editing the `print_tabular` method to print
+different attributes of the Nodes in the Graph. (For example, we might
+want to see the Node’s `input_nodes` and `users`.)
+
+(available-debuggers)=
```

Comment:
This has to be the same as in the .rst "Available Debuggers" as it fails the doc [build](https://github.com/pytorch/pytorch/actions/runs/15523241020/job/43743427465?pr=155429#step:11:876) - need to fix for all targets. 
```suggestion
(available-debuggers)=
```